extensions/amp-access/0.1/amp-access.js

```diff
@@ -185,6 +186,28 @@ export class AccessService {
         this.performance_.flush();
       }
     });
+
+    // Re-authorize newly added sections.
+    ampdoc.getRootNode().addEventListener(AmpEvents.DOM_UPDATE,
+        this.onDomUpdate_.bind(this));
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  onDomUpdate_(event) {
+    // Only re-authorize sections if the response is already available.
+    // Otherwise, just wait for the authorization - it will cover new sections.
+    // But wait for the last authorization operation to complete.
+    const response = this.authResponse_;
+    if (response) {
+      return this.lastAuthorizationPromise_.then(() => {
+        const target = dev().assertElement(event.target);
+        this.applyAuthorizationToRoot_(target,
+            /** @type {!JsonObject} */ (response));
```

Comment:
@erwinmombay I added `!JsonObject` here to remove `|null` to satisfy compiler. But I have no idea why it's needed. Seems completely wrong. I already check `if (response)` above. Why do I need this here?